python_modules/dagster/dagster/core/execution/plan/execute_plan.py

```diff
@@ -45,73 +45,89 @@ def inner_plan_execution_iterator(
-                    yield DagsterEvent.engine_event(
-                        pipeline_context=pipeline_context,
-                        message="Exception while setting up compute log capture",
-                        event_specific_data=EngineEventData(
-                            error=serializable_error_info_from_exc_info(sys.exc_info())
-                        ),
-                        step_handle=step_context.step.handle,
-                    )
-                    log_capture_error = e
-
-                if not log_capture_error:
-                    yield DagsterEvent.capture_logs(
-                        step_context, log_key=step_context.step.key, steps=[step_context.step]
-                    )
-
-                for step_event in check.generator(dagster_event_sequence_for_step(step_context)):
-                    check.inst(step_event, DagsterEvent)
-                    step_event_list.append(step_event)
-                    yield step_event
-                    active_execution.handle_event(step_event)
-
-                active_execution.verify_complete(pipeline_context, step.key)
-
-                try:
-                    stack.close()
-                except Exception:
-                    yield DagsterEvent.engine_event(
-                        pipeline_context=pipeline_context,
-                        message="Exception while cleaning up compute log capture",
-                        event_specific_data=EngineEventData(
-                            error=serializable_error_info_from_exc_info(sys.exc_info())
-                        ),
-                        step_handle=step_context.step.handle,
-                    )
-
-            # process skips from failures or uncovered inputs
-            for event in active_execution.plan_events_iterator(pipeline_context):
-                step_event_list.append(event)
-                yield event
-
-            # pass a list of step events to hooks
-            for hook_event in _trigger_hook(step_context, step_event_list):
-                yield hook_event
+
+                # process skips from failures or uncovered inputs
+                for event in active_execution.plan_events_iterator(pipeline_context):
+                    step_event_list.append(event)
+                    yield event
+
+            except GeneratorExit:
+                # Shouldn't happen, but avoid runtime-exception in case this generator gets GC-ed
+                # (see https://amir.rachum.com/blog/2017/03/03/generator-cleanup/).
+                generator_closed = True
+                raise
+            except (KeyboardInterrupt, DagsterExecutionInterruptedError):
+                interrupted = True
+                raise
+            finally:
+                if not generator_closed and not interrupted:
+                    # pass a list of step events to hooks
+                    for hook_event in _trigger_hook(step_context, step_event_list):
+                        yield hook_event
```

Comment:
this looks worse than it is, all i really did here was add a try:finally: block and move the hook code into the finally

Reply:
what about the variant where we just don't raise immediately on unknown exception types?